common/djangoapps/third_party_auth/middleware.py

```diff
@@ -23,3 +25,23 @@ def get_redirect_uri(self, request, exception):
             redirect_uri = pipeline.AUTH_DISPATCH_URLS[auth_entry]
 
         return redirect_uri
+
+
+class PipelineQuarantineMiddleware(object):
+
+    def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
+        """
+        Check the session to see if we've quarantined the user to a particular
+        step of the authentication pipeline; if so, look up which modules the
+        user is allowed to browse to without breaking the pipeline. If the view
+        that's been requested is outside those modules, then flush the session.
```

Comment:
This describes what the code does, quite clearly, but doesn't really say why, no what parts of the codebase will be using this feature.